course5/homework12/homework12.cc

```diff
@@ -0,0 +1,138 @@
+#include "comparers.h"
+#include <vector>
+#include <algorithm>
+#include <string>
+#include <iostream>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <stdexcept>
+#include <fstream>
+#include <boost/program_options/options_description.hpp>
+#include <boost/program_options/variables_map.hpp>
+#include <boost/program_options/parsers.hpp>
+#include <boost/tokenizer.hpp>
+
+namespace po = boost::program_options;
+namespace homework12{
+
+po::variables_map vm;
+
+po::options_description DeclareOptions()
+{
+    po::options_description desc;
+    desc.add_options()
+            ("help", "produce help message")
+            ("reverse,r", "reverse the result of comparisons")
+            ("numeric-sort,d", "compare according to string numerical value")
+            ("key,c", po::value<unsigned int>(), "sort via a key; value specifies column number")
+            ("field-separator,s", po::value<char>()->default_value('/n'), "use SEP instead of non-blank to blank transition")
+            ("input,i", po::value<std::string>(), "input file, default stdin")
+            ("output,o", po::value<std::string>(), "output file, default stdout");
+    return desc;
+}
+
+po::variables_map ReadOptions(int argc, char** argv, const po::options_description& desc)
+{
+    po::variables_map vm;
+    po::store(po::parse_command_line(argc, argv, desc), vm);
+    po::notify(vm);
+
+    return vm;
+}
+}
+
+std::vector<std::string> ReadLines(std::istream&  file)
```

Comment:
Mark all your functions except of ```main``` as ```static```. This will inform the compiler that no linker symbols have to be generated for the, as they are not used outside of the module and don't have to participate in linking. 
This will also give the compiler more optimization opportunities.